Add page entries info to serials page title to distinguish titles

myokoym opened the pull request:
Close https://github.com/bm-sms/nursepress/issues/284

Commit:
Add page entries info to serials page title to distinguish titles

okkez commented:
Looks nice :sunglasses:

tricknotes commented on app/decorators/site_decorator.rb:
```diff
@@ -34,4 +34,13 @@ def page_title(posts)
     end
   end
+
+  def serials_page_title(serials)
```
Can you create `SerialsDecorator` and move `serials_page_title` to it?
The `serials_page_title` method doesn't use the attributes of `Site`, so this method shouldn't exist on `SiteDecorator`.

myokoym replied:
It's reasonable. I'll create `SerialsDecorator`.

myokoym replied:
I've fixed it by 3f15bdae03863425419db94e6c5599f1a22e6f2a .

Commit:
Extract SerialsDecorator

Because serials_page_title doesn't use the site object.

tricknotes commented on app/controllers/serials_controller.rb:
```diff
@@ -1,6 +1,8 @@
   def index
     @serials = current_site.serials.published.order(id: :desc).page(params[:page])
+    @serials.extend(PaginationInfoDecorator)
+    @serials.extend(SerialsDecorator)
```
I think `SerialsDecorator` depends on `PaginationInfoDecorator`. ( https://github.com/bm-sms/daimon-news/pull/551/files#diff-f7ddda9786b3de2ac334a4bc6f8d5751R5 👀 )
Can you include `PaginationInfoDecorator` into `SerialsDecorator`?

myokoym replied:
I understand. I've fixed it.

cafedomancer commented on app/controllers/serials_controller.rb:
```diff
@@ -1,6 +1,8 @@
 class SerialsController < ApplicationController
   def index
     @serials = current_site.serials.published.order(id: :desc).page(params[:page])
```
By the way, it is not related to this pull request, but what is `order(id: :desc)`? I don't know its intention from this code.

okkez replied:
In SQL specification, unordered query returns unsorted result possibly.
So we have to specify order, if we need stable sorted result.

cafedomancer replied:
OK. Give it an appropriate name as `scope`? It's the problem that we don't understand that from `order(id: :desc)`.

okkez replied:
- Pagination needs sorted list (result of query)
- Seiral model does not have `order` column

---

:memo: 連載は新しい順に並んでるのが自然だと思う。

tricknotes replied:
We discussed on chat about this topic.
We decided name this order specification as scope and use it here.
- To specify order for pagination.
- To be meaningful for programmer.

Commit:
Include PaginationInfoDecorator for SerialsDecorator

Because SerialsDecorator.#page_title depends on
PaginationInfoDecorator.#page_entries_info.

tricknotes commented:
Look awesome!
Thanks for your nice fix @myokoym 🚢
